jina/drivers/index.py

```diff
@@ -39,6 +40,28 @@ class KVIndexDriver(BaseIndexDriver):
     """
 
     def _apply_all(self, docs: Iterable['jina_pb2.Document'], *args, **kwargs) -> None:
-        keys = [uid.id2hash(doc.id) for doc in docs]
-        values = [doc.SerializeToString() for doc in docs]
+        keys = []
+        values = []
+        for doc in docs:
+            if doc.indexed:
+                continue
+            keys.append(uid.id2hash(doc.id))
+            values.append(doc.SerializeToString())
```

Comment:
list comprehension, `[... for doc in docs if doc.tags.get('is_indexed', False)]`